doc/run-cloud-build-locally.md

```diff
@@ -0,0 +1,197 @@
+# Running `cloudbuild.yaml` Locally with Docker
+
+`cloudbuild-local` is deprecated; however you can still iterate on the
+`cloudbuild.yaml` steps by running them manually using Docker and the same
+tooling that Cloud Build uses. This document provides a lightweight, repeatable
+workflow for running the CI pipeline locally while being able to iterate
+quickly.
+
+## Prerequisites
+
+- Docker Desktop (with `docker compose`) installed and running.
+- A local JDK 17 + Maven 3.8.x for the Maven build steps.
+- `python3` (for `synthea-uploader` unit tests) and `python -m unittest`
+  available on `$PATH`.
+- Access to the workspace root (same directory that holds `cloudbuild.yaml`).
+
+## Environment setup
+
+1. Create an external Docker network named `cloudbuild`:
+
+```bash
+docker network create cloudbuild
+```
+
+2. From the repo root, export substitutions that Cloud Build normally injects:
+
+```bash
+export _TAG=local
+export _REPOSITORY=fhir-analytics
+```
+
+3. If you need to mirror the GCP `gcloud` network configuration (Docker Compose
+   services talk to each other by name), run all commands inside this directory
+   so the Compose files resolve relative paths correctly.
+
+## Step-by-step local reproduction
+
+### 1. Start shared services
+
+Start the HAPI source and both sink servers the same way Cloud Build does:
+
+```bash
+docker compose -f docker/hapi-compose.yml -p hapi-compose up --force-recreate --remove-orphans -d
+
+SINK_SERVER_NAME=sink-server-search SINK_SERVER_PORT=9001 docker compose -f docker/sink-compose.yml -p sink-server-search up --force-recreate --remove-orphans -d
+
+SINK_SERVER_NAME=sink-server-jdbc SINK_SERVER_PORT=9002 docker compose -f docker/sink-compose.yml -p sink-server-jdbc up --force-recreate --remove-orphans -d
+```
+
+Wait for the servers to become ready and confirm via the command:
+
+```bash
+docker ps
+```
+
+You should get output similar to:
+
+```
+CONTAINER ID   IMAGE                     COMMAND                  CREATED              STATUS              PORTS                    NAMES
+3a23504297d0   hapiproject/hapi:latest   "java --class-path /…"   About a minute ago   Up About a minute   0.0.0.0:9002->8080/tcp   sink-server-jdbc
+c13ad0618e65   hapiproject/hapi:latest   "java --class-path /…"   2 minutes ago        Up 2 minutes        0.0.0.0:9001->8080/tcp   sink-server-search
+1f301480d87f   hapiproject/hapi:latest   "java --class-path /…"   15 minutes ago       Up 15 minutes       0.0.0.0:8091->8080/tcp   hapi-server
+815bc05a6005   postgres                  "docker-entrypoint.s…"   15 minutes ago       Up 15 minutes       0.0.0.0:5432->5432/tcp   hapi-fhir-db
+```
+
+### 2. Compile the Java modules
+
+```bash
+mvn --no-transfer-progress -e -T 2C install -Dlicense.skip=true -Dspotless.apply.skip=true
+```
+
+### 3. Upload sample synthetic data to HAPI stored in
```

Comment:
"stored in ?" I think the title and first sentence below are connected (and need to be separated).